keyvault/client/src/main/java/com/azure/keyvault/SecretRequestAttributes.java

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.keyvault;
+
+import com.azure.keyvault.models.SecretAttributes;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+/**
+ * The object attributes managed by the Secret service.
+ */
+class SecretRequestAttributes {
+
+    /**
+     * Creates an instance of SecretRequestAttributes. Reads secretAttributes.notBefore, secretAttributes.expires and secretAttributes.enabled fields
+     * from {@code secretAttributes}
+     * @param secretAttributes the {@link SecretAttributes} object with populated attributes
+     */
+    SecretRequestAttributes(SecretAttributes secretAttributes) {
+        if (secretAttributes.notBefore() != null) {
+            this.notBefore = secretAttributes.notBefore().toEpochSecond();
+        }
+        if (secretAttributes.expires() != null) {
+            this.expires = secretAttributes.expires().toEpochSecond();
+        }
+        this.enabled = secretAttributes.enabled();
+    }
+
+    /**
+     * The secret value.
+     */
+    @JsonProperty(value = "value")
+    private String value;
+
+    /**
+     * The secret id.
+     */
+    @JsonProperty(value = "id")
+    private String id;
+
+    /**
+     * Determines whether the object is enabled.
+     */
+    @JsonProperty(value = "enabled")
+    private Boolean enabled;
```

Comment:
I'm curious as to whether we prefer the object, `Boolean` or the primitive `boolean` by default. Is there some coding style for this, @JonathanGiles? Similarly, for the `Long` below.

Reply:
I generally dislike the use of reference types when there exists a primitive type. The argument to use reference types in some cases is to differentiate between the existence of a value and the non existence of a value, due to the ability to reason that a null value represents the non existence of that value, which is hard to represent in primitives.

Reply:
@g2vinay Given this feedback, can you please consider whether you need to use reference types Boolean and Long?

Reply:
It is less of an issue in general here than in other places, because this is not public API, but still, it is a good place for bugs to creep in. Reference types are slower, more error prone (because null exists), and also can lead to correctness issues (e.g. `new Integer(42) == new Integer(42)` returns false!)

Reply:
Introducing primitives will create issues.
The enabled, notBefore, expires fields are optional.
If not specified by user, primitives will assign them default values.
The default values for boolean and long will assign erroneous/wrong/unintentional values to the variables.
boolean enabled variable will become false by default.
long notBefore and expiry will become something like 1970-01-01TAA:BB:CCZ in UTC time.